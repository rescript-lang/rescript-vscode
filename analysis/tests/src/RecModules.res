module rec A: {
  type t

  @send external child: t => B.t = "child"
} = A

and B: {
  type t

  @send external parent: t => A.t = "parent"
} = B

module C = {
  type t

  @send external createA: t => A.t = "createA"
}

module MC = C
//          ^hov
module MA = A
//          ^hov
