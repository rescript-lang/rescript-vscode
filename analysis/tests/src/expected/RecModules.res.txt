Hover src/RecModules.res 18:12
{"contents": {"kind": "markdown", "value": "```rescript\nmodule C: {\n  type t\n  let createA: t => A.t\n}\n```"}}

Hover src/RecModules.res 20:12
{"contents": {"kind": "markdown", "value": "```rescript\nmodule A: {\n  type t\n  let child: t => B.t\n}\n```"}}

